server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -282,6 +282,18 @@ public final void validate(final Settings settings, final boolean validateDepend
         validate(settings, validateDependencies, false, false);
     }
 
+    /**
+     * Validates that all settings are registered and valid.
+     *
+     * @param settings              the settings to validate
+     * @param validateDependencies  true if dependent settings should be validated
+     * @param validateInternalIndex true if internal index settings should be validated
+     * @see Setting#getSettingsDependencies(String)
+     */
+    public final void validate(final Settings settings, final boolean validateDependencies, final boolean validateInternalIndex) {
```

Comment:
Think it might be better to either make `validateInternalIndex` more explicit and be `validateInternalIndexSettings` or remove "Index" and have it as `validateInternal` or `validateInternalSettings`. The reason is that at first glance this looks like its a flag for whether we validate settings in "internal indexes".

Reply:
Do you think that this isn't clear enough from the provided Javadocs? `validateInternalIndexSettings` is such a long name and it would cause some lines to spill over the line length limit. I am happy to adjust if you feel strongly, but I want to double check before making the change because of the reformatting it will require.

Reply:
I think its ok when you read the JavaDoc but it just seems a bit strange when first reading the variable name. I agree that `validateInternalIndexSettings ` is too long. Are you against making it `validateInternalSettings` instead?

Reply:
I am hesitant on `validateInternalSettings` because we do not have a general notion of internal settings, only internal index settings for now.

Reply:
Ok, thats true. In which case I'm ok with leaving this as is.